### DISCUSSION TITLE
What's the relationship between this code-server and Microsoft's 'VS Code Server' service?

### DISCUSSION THREAD

**Discussion by liuyiqi1999** [coder/code-server#5517] [Q&A] [answered]
Microsoft released a 'VS Code Server' service recently. According to the [doc](https://code.visualstudio.com/docs/remote/vscode-server) , it looks basically the same as what this code-server does. It's CLI tool is even also called 'code-server'. 

I'm wondering what's the relationship between these two projects, and if not related, what will this code-server going to be if the Microsoft VS Code Server's no longer 'private preview' any more.

**Comment by jsjoeio** [comment, marked as answer]
Great question! It does indeed seem like the two projects are serving the same purpose. Is theirs still in private preview?

code-server (ours) can be run behind a proxy. Not sure if theirs can. We'll have to wait and see. I know that's a little vague but we don't want to say too much until their project is a little bit more mature.

**Reply by lg-nineoneninethreenine** [reply]
Yeah, but ours is much more faster, right? And according to me, it actually used the shell configuration. I have a circuit problem on my keyboard and the G9 key is always pressed. I blocked it through my shell config but VSCode doesn't accept it. code-server does, which helps me.

**Reply by jsjoeio** [reply]
Ah, that's good to hear! Yeah, I'm sure there are other things we could list as well. Only time will tell!

**Reply by lg-nineoneninethreenine** [reply]
Oh it was F9 sorry. But I would prefer _this_ than anything. I mean, this is a better VS Code that is more faster.

**Reply by lg-nineoneninethreenine** [reply]
code-server by VS Code is _stupid_. Doesn't work on my PC whereas code-server works, so win for us.
